### PR TITLE
Implemented schematron <value-of select="..."/>.

### PR DESCRIPTION
**Pull request by nverwer**
Also:
Moved message formatting into FormatterBase class.
Added error message for abstract rules that are not found.

Message formatting still uses regular expressions instead of the XML structure.

### CODE REVIEW

**Comment by kzu** [comment]
Heya, thanks for this!

I've just uploaded a .editorconfig that should make things easier to keep consistent (formatting-wise) and easier to review. Please install the EditorConfig extension in VS and that should be all you need to do to get the right tabs/spaces.

Also, could you add some unit tests that verify the new behavior? They typically are very useful to also figure out what is implemented and how it works in practice ;)

Thanks!

**Comment by nverwer** [comment]
Hello Daniel,

I will have a look at the layout and add some unit tests soon.

**Comment by kzu** [comment]
Great!

File: src/Schematron.Tests/Schematron.Tests.csproj [hunk 1]
```diff
@@ -86,6 +86,9 @@
       <Name>Schematron</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (nverwer) [inline src/Schematron.Tests/Schematron.Tests.csproj:90]: Apparently, this makes the project into a unit testable project.

File: src/Schematron.Tests/ValidatorTests.cs [hunk 2]
```diff
@@ -8,121 +8,149 @@
 using Xunit;
 
 namespace Schematron.Tests
-{
```
Comment (nverwer) [inline src/Schematron.Tests/ValidatorTests.cs:11]: Formatting was changed by EditorConfig.

**Comment by nverwer** [comment]
I hope that the lay-out is okay now. There is a unit test now.

**Comment by kzu** [comment]
Looks great, thanks!

**Pull request merged**